collections/src/lib.rs

```diff
@@ -32,16 +32,61 @@ fn mode<T: Eq + Copy + Hash + Ord>(it: &[T]) -> T {
 use std::collections::HashSet;
 use std::iter::FromIterator;
 
-fn to_pig_latin(word: &mut String) {
-    let VOWELS: HashSet<char> = HashSet::from_iter("aeiou".chars());
-    if !VOWELS.contains(&word.char_indices().next().unwrap().1.to_ascii_lowercase()) {
+pub fn to_pig_latin(word: &mut String) {
+    let vowels: HashSet<char> = HashSet::from_iter("aeiou".chars());
+    if !vowels.contains(&word.char_indices().next().unwrap().1.to_ascii_lowercase()) {
         let ch = word.remove(0);
         word.push_str(&format!("-{}ay", ch.to_string()));
     } else {
         word.push_str("-hay");
     }
 }
 
+fn sorted<T: Ord + Clone>(collection: &[T]) -> Vec<T> {
+    let mut copy = collection.to_owned();
+    copy.sort();
+    copy
+}
+
+use regex::Regex;
+use std::collections::BTreeMap;
+// We use btree and not hashmap to keep the keys sorted.
+type Employees = BTreeMap<String, Vec<String>>;
+// Instead of printing I return a string with what should be printed.
+pub fn process_command(employees: &mut Employees, command: &str) -> Option<String> {
+    let add_command = Regex::new(r"(?i)add (?P<name>\w+) to (?P<department>\w+)").unwrap();
+    let list_command = Regex::new(r"(?i)list(?:\s(?P<department>\w+))?").unwrap();
+    if let Some(add) = add_command.captures(&command.to_ascii_lowercase()) {
+        employees
+            .entry(add.name("department").unwrap().as_str().to_owned())
+            .or_default()
+            .push(add.name("name").unwrap().as_str().to_owned());
+        None
+    } else if let Some(list_match) = list_command.captures(command) {
+        let format_department = |department: &str, names: &[String]| {
```

Comment:
why define a named function using an anonymous function?
```suggestion
        fn format_department(department: &str, names: &[String]) {
```

Reply:
Thought I couldn't do an inner function - turns out I was wrong!